examples/field_examples/viking_graben_line12/rtm/rtm.jl

```diff
@@ -0,0 +1,188 @@
+# using Pkg
+# Pkg.add("Statistics")
+# Pkg.add("Random")
+# Pkg.add("LinearAlgebra")
+# Pkg.add("Interpolations")
+# Pkg.add("DelimitedFiles")
+# Pkg.add("Distributed")
+# Pkg.add("DSP")
+# Pkg.add("SlimOptim")
+# Pkg.add("NLopt")
+# Pkg.add("HDF5")
+# Pkg.add("SegyIO")
+# Pkg.add("Plots")
+# Pkg.add("SetIntersectionProjection")
+
+using Statistics, Random, LinearAlgebra, Interpolations, DelimitedFiles, Distributed, DSP
+using JUDI, NLopt, HDF5, SegyIO, Plots
+
+############# INITIAL DATA #############
+modeling_type = "acoustic"  # scalar, acoustic
+
+prestk_dir = "$(@__DIR__)/../proc/"
+prestk_file = "s_deghost_gain_mute_dip_radon.sgy"
+
+dir_out = "$(@__DIR__)/"
+
+# set frequency in that path to get the latest updated model
+model_file = "$(@__DIR__)/../fwi/fwi_$(modeling_type)/0.005Hz/model.h5"
+
+# use original wavelet file 
+# wavelet_file = "$(@__DIR__)/../FarField.dat" # dt=1, skip=25
+# or use deghosted wavelet
+wavelet_file = "$(@__DIR__)/../proc/FarField_deghosted.dat" # dt=4, skip=0
+wavelet_skip_start = 0 # 25 [lines] for raw source and 0 for deghosted source
+wavelet_dt = 4          # 1 [ms] for raw source and 4 [ms] for deghosted source
+
+segy_depth_key_src = "SourceSurfaceElevation"
+segy_depth_key_rec = "RecGroupElevation"
+
+seabed = 355  # [m]
+
+# water velocity, km/s
+global vwater = 1.5
+
+# water density, g/cm^3
+global rhowater = 1.02
+
+# JUDI options
+buffer_size = 0f0    # limit model (meters)
+
+# prepare folder for output data
+mkpath(dir_out)
+
+# Load data and create data vector
+# block = segy_read(prestk_dir * prestk_file)
+container = segy_scan(prestk_dir, prestk_file, ["SourceX", "SourceY", "GroupX", "GroupY", "RecGroupElevation", "SourceSurfaceElevation", "dt"])
+d_obs = judiVector(container; segy_depth_key = segy_depth_key_rec)
+
+srcx = Float32.(get_header(container, "SourceX")[:,1])
+grpx = Float32.(get_header(container, "GroupX")[:,1])
+min_src_x = minimum(srcx)./1000f0
+max_src_x = maximum(srcx)./1000f0
+min_grp_x = minimum(grpx)./1000f0
+max_grp_x = maximum(grpx)./1000f0
+min_cdp_x = (min_src_x+min_grp_x)/2f0
+max_cdp_x = (max_src_x+max_grp_x)/2f0
+min_x = minimum([min_src_x, min_grp_x])
+max_x = maximum([max_src_x, max_grp_x])
+
+# helpful functions
+rho_from_slowness(m) = 0.23.*(sqrt.(1f0 ./ m).*1000f0).^0.25
```

Comment:
That's Gardner right? Maybe name it so

Reply:
You can also put utilities that are used in multiple places (like fwi/rtm) in an `utils.jl`. DrWatson.jl can helo with that too.

Reply:
Right